airflow/models.py

```diff
@@ -137,13 +137,13 @@ def clear_task_instances(tis, session, activate_dag_runs=True, dag=None):
             if dag and dag.has_task(task_id):
                 task = dag.get_task(task_id)
                 task_retries = task.retries
-                ti.max_tries = ti.try_number + task_retries
+                ti.max_tries = ti._try_number + task_retries
```

Comment:
Why use _try_number here?

Reply:
No great reason - other than otherwise it would be `ti.try_number - 1 + task_retries` which isn't the most clear. I can go with what so that `_try_number` is then never accessed from outside the TI class.